lasagne/layers/conv.py

```diff
@@ -16,6 +17,54 @@
 
 def conv_output_length(input_length, filter_size,
                        stride, border_mode, pad=0):
+    """Helper function to compute the output size of a convolution operation
+
+    This function computes the length along a single axis, which corresponds
+    to a 1D convolution. It can also be used for convolutions with higher
+    dimensionalities by using it individually for each axis.
+
+    Parameters
+    ----------
+    input_length : int
+        The size of the input.
+
+    filter_size : int
+        The size of the filter.
+
+    stride : int
+        The stride of the convolution operation.
+
+    border_mode : str, 'valid', 'full', 'same' or 'pad'
+        A string indicating the convolution border mode.
+
+        If 'valid', it is assumed that the convolution is only computed where
+        the input and the filter fully overlap.
+
+        If 'full', it is assumed that the convolution is computed wherever the
+        input and the filter overlap by at least one position.
+
+        If 'same', it is assumed that the convolution is computed wherever the
+        input and the filter overlap by half, which results in an output length
+        that is the same as the input length.
```

Comment:
ah yes, this is coming back to me. I'd argue that it doesn't actually do the correct thing at the moment (it trims a `full` convolution, which gives a result that's not symmetric for even filters).

But either way the current wording implies that with a filter and an input of length 2, you'd get an output of length 3.